service/storage_service.cc

```diff
@@ -5627,7 +5629,11 @@ future<raft_topology_cmd_result> storage_service::raft_topology_cmd_handler(raft
                                 parent_info.id, streaming::stream_reason::rebuild, _rebuild_result, [this, &source_dc] () -> future<> {
                             auto tmptr = get_token_metadata_ptr();
                             if (is_repair_based_node_ops_enabled(streaming::stream_reason::rebuild)) {
-                                co_await _repair.local().rebuild_with_repair(tmptr, std::move(source_dc));
+                                utils::optional_param sdc_param;
+                                if (!source_dc.empty()) {
+                                    sdc_param.emplace(source_dc).set_flag(utils::optional_param::flag::user_provided);
```

Comment:
We could have `mark_user_provided()` to simplify this.

Reply:
okay, this can be added quickly.